### DISCUSSION TITLE
ggml-cuda catastrophic error : out of memory

### DISCUSSION THREAD

**Discussion by jygmysoul** [ggml-org/llama.cpp#10337] [Q&A]
My previous July version of LLAMACPP was compiled successfully. The system environment is WIN10+RTX3080+10G, and the compilation can be completed normally. This time I want to update to the latest version of LLAMACPP, but the compilation error is reported, and the prompt is as follows:
llama.cpp-b4095\ggml\src\ggml-cuda\common.cuh(392): catastrophic error : out of memory

I did not change any compilation parameters, only changed
GGML_CUDA=1, does the latest version have such high hardware requirements when compiling?
What hardware environment does the new version require? In addition, can I reduce the requirement for video memory capacity at compile time by modifying the configuration?

**Comment by wooooyeahhhh** [comment]
the argument is `LLAMA_CUDA=1` not `GGML_CUDA=1`

**Comment by jygmysoul** [comment]
CUDA
This provides GPU acceleration using the CUDA cores of your Nvidia GPU. Make sure to have the CUDA toolkit installed. You can download it from your Linux distro's package manager (e.g. apt install nvidia-cuda-toolkit) or from here: [CUDA Toolkit](https://developer.nvidia.com/cuda-downloads).

For Jetson user, if you have Jetson Orin, you can try this: [Offical Support](https://www.jetson-ai-lab.com/tutorial_text-generation.html). If you are using an old model(nano/TX2), need some additional operations before compiling.

Using make:

make GGML_CUDA=1
Using CMake:

cmake -B build -DGGML_CUDA=ON
cmake --build build --config Release
The environment variable [CUDA_VISIBLE_DEVICES](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#env-vars) can be used to specify which GPU(s) will be used.
=================================
LLAMA_CUDA=1 is the parameter of the old version
GGML_CUDA=1 is the parameter of the new version

Waiting for official answer

**Comment by wooooyeahhhh** [comment]
I can complete compilation with worse hardware so it can't be a hardware issue
